catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java

```diff
@@ -0,0 +1,153 @@
+                            // Check for warnings and errors
+                            if (validationErrorsExist) {
+                                validationErrors.add(getValidatorName(metacardValidator));
+                            }
+                            if (validationWarningsExist) {
+                                validationWarnings.add(getValidatorName(metacardValidator));
+                            }
+                        } else {
+                            LOGGER.error(
+                                    "Metacard validator {} did not have any warnings or errors but it threw a validation exception."
+                                            + " There is likely something wrong with your implementation. This will result in the metacard not"
+                                            + " being properly marked as invalid.",
+                                    getValidatorName(metacardValidator));
+                        }
+
+                    } else {
+                        // If validator is explicitly turned on, break out and do not include in the
+                        // list of metacards that will be allowed through.
+                        toReturn = false;
+                        break;
+                    }
+
+                }
+            }
+            if (toReturn) {
+                newMetacard
+                        .setAttribute(new AttributeImpl(VALIDATION_WARNINGS, validationWarnings));
+                newMetacard.setAttribute(new AttributeImpl(VALIDATION_ERRORS, validationErrors));
+                returnMetacards.add(newMetacard);
+            }
+        }
+        return new CreateRequestImpl(returnMetacards, input.getProperties());
+    }
+
+    @Override
+    public UpdateRequest process(UpdateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        return input;
+    }
+
+    @Override
+    public DeleteRequest process(DeleteRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        return input;
+    }
+
+    public void setEnforcedMetacardValidators(List<String> enforcedMetacardValidators) {
+        this.enforcedMetacardValidators = enforcedMetacardValidators;
+    }
+
+    public List<String> getEnforcedMetacardValidators() {
+        return this.enforcedMetacardValidators;
+    }
+
+    public void setMetacardValidators(List<MetacardValidator> metacardValidators) {
+        this.metacardValidators = metacardValidators;
+    }
+
+    public List<MetacardValidator> getMetacardValidators() {
+        return this.metacardValidators;
+    }
+
+    private Boolean checkEnforcedMetacardValidators(MetacardValidator metacardValidator) {
+        return (null == enforcedMetacardValidators || !enforcedMetacardValidators
+                .contains(getValidatorName(metacardValidator)));
+    }
+
+    private static String getValidatorName(MetacardValidator metacardValidator) {
```

Comment:
It doesn't depend on the MetacardValidityMarkerPlugin at all, so it didn't really make sense to make it non-static. It could possibly go on the MetacardValidator interface, but then it would have to mess with already existing code and possibly interfere with backwards compatibility.